src/features/mcda/atoms/mcdaCalculation.ts

```diff
@@ -0,0 +1,223 @@
+import isEqual from 'lodash/isEqual';
+import sum from 'lodash/sum';
+import { createAtom } from '~utils/atoms';
+import { layersRegistryAtom } from '~core/logical_layers/atoms/layersRegistry';
+import { layersSettingsAtom } from '~core/logical_layers/atoms/layersSettings';
+import { createUpdateLayerActions } from '~core/logical_layers/utils/createUpdateActions';
+import {
+  anyCondition,
+  featureProp,
+  notEqual,
+} from '~utils/bivariate/bivariate_style/styleGen';
+import { adaptTileUrl } from '~utils/bivariate/tile/adaptTileUrl';
+import appConfig from '~core/app_config';
+import {
+  MCDARenderer,
+  sentimentDefault,
+  sentimentReversed,
+} from '../renderers/MCDARenderer';
+import type { BivariateLayerStyle } from '~utils/bivariate/bivariateColorThemeUtils';
+import type { Quotient } from '~utils/bivariate';
+
+export interface JsonMCDA {
+  id?: string;
+  axes: [string, string][];
+  ranges: [number, number][];
+  sentiments: [string, string][];
+  coefficients: number[];
+  colors: {
+    good: string;
+    bad: string;
+  };
+}
+
+const DEFAULT_GREEN = 'rgba(90, 200, 127, 0.5)';
+const DEFAULT_RED = 'rgba(228, 26, 28, 0.5)';
+
+const filterSetup = (quotients: Quotient[]) =>
+  anyCondition(
+    ...quotients.map((quotient) =>
+      notEqual(['/', featureProp(quotient[0]), featureProp(quotient[1])], 0),
+    ),
+  );
+
+const layerNormalized = (
+  [num, den]: Quotient,
+  [min, max]: [number, number],
+  sentiment: [string, string],
+  coefficient: number,
```

Comment:
need renormalization somewhere (divide each one by sum of all in the layer) so one can change just one indicator value in the formula.

Also, have a look at RVA Editor formula field for inspiration.

Reply:
Hm, we do it already here: https://github.com/konturio/disaster-ninja-fe/pull/437/files#diff-87410fd4a4545628551fda85f2927a65f9bd7a5dcd02810f7bec3f8dfbe5f96cR89

So it's implemented as it's in task description formulas: (https://kontur.fibery.io/Tasks/Task/Add-MCDA-layers-constructor-on-DN-FE-14520)
V_norm = (V_current - V_min)/(V_max - V_min)
V_total = Σ(V_norm(1)*weight(1), V_norm(2)*weight(2), V_norm(n)*weight(n))/Σweight

Could you please explain what you would like to introduce in this way?

Reply:
I did not find this logic when reading this PR, if it exists then please disregard.